paths/paths.go

```diff
@@ -3,11 +3,26 @@ package paths
 import (
 	"os"
 	"path/filepath"
+
+	logging "github.com/ipfs/go-log"
+	"github.com/mitchellh/go-homedir"
 )
 
+var log logging.EventLogger
+var defaultHomeDir string
+
+func init() {
+	log = logging.Logger("paths")
+	var err error
+	defaultHomeDir, err = homedir.Expand(defaultHomeDirUnexpanded)
+	if err != nil {
+		log.Errorf("could not expand ~/.filecoin default home directory: %s", err)
```

Comment:
Panic. We are not in a valid configuration if this check fails. Either we panic or we move this error check someplace where we can shutdown correctly.